shared/fs/row/row.js

```diff
@@ -1,39 +1,85 @@
 // @flow
 import * as React from 'react'
 import * as Types from '../../constants/types/fs'
-import {globalStyles, globalColors, globalMargins, isMobile} from '../../styles'
+import {globalStyles, globalColors, globalMargins, isMobile, glamorous} from '../../styles'
 import memoize from 'lodash/memoize'
 import {Box, ClickableBox, Icon, Text, Divider} from '../../common-adapters'
 import PathItemIcon from './path-item-icon'
+import PathItemInfo from './path-item-info'
 
 type RowProps = {
   name: string,
   type: Types.PathType,
+  lastModifiedTimestamp: number,
+  lastWriter: string,
   itemStyles: Types.ItemStyles,
   onOpen: () => void,
   openInFileUI: () => void,
+  onAction: (event: SyntheticEvent<>) => void,
 }
 
-export const Row = ({name, type, itemStyles, onOpen, openInFileUI}: RowProps) => (
+const HoverBox = isMobile
+  ? Box
+  : glamorous(Box)({
+      '& .fs-path-item-hover-icon': {
+        color: globalColors.white,
+      },
+      ':hover .fs-path-item-hover-icon': {
+        color: globalColors.black_40,
+      },
+      '& .fs-path-item-hover-icon:hover': {
+        color: globalColors.black_60,
+      },
+    })
+
+export const Row = ({
+  name,
+  type,
+  lastModifiedTimestamp,
+  lastWriter,
+  itemStyles,
+  onOpen,
+  openInFileUI,
+  onAction,
+}: RowProps) => (
   <Box>
-    <ClickableBox onClick={onOpen} style={stylesCommonRow}>
-      <Box style={stylesRowContainer}>
-        <Box style={stylesRowBox}>
-          <PathItemIcon spec={itemStyles.iconSpec} />
+    <Box style={stylesCommonRow}>
+      <HoverBox style={stylesRowContainer}>
+        <ClickableBox onClick={onOpen} style={stylesRowBox}>
+          <PathItemIcon spec={itemStyles.iconSpec} style={pathItemIconStyle} />
           <Box style={folderBoxStyle}>
-            <Text type={itemStyles.textType} style={rowTextStyles(itemStyles.textColor)}>
+            <Text
+              type={itemStyles.textType}
+              style={rowTextStyles(itemStyles.textColor)}
+              lineClamp={isMobile ? 1 : undefined}
+            >
               {name}
             </Text>
+            {type !== 'folder' ? (
```

Comment:
just do `type !== 'folder' && <PathItemInfo />`

Reply:
Oh hmm somehow I was under the impression that this would cause react to render a `false` (since that expression evaluates to `false`). But I just tried it and it does not!

What does React use to determine whether a thing is rendered or skipped?

Reply:
If it's falsey it'll not render

Reply:
Hmm I just put a `{true}` there and it didn't render either. I guess type also matters somehow.

Anyway thanks for pointing it out!